slices/libpam-runtime.yaml

```diff
@@ -18,28 +18,125 @@ slices:
   # pam-config and pam-defaults slices are only used for generation, and are useless
   # for anything else.
 
-  # Applications that need libpam for now should just rely on the config and var slices for now -
-  # until appropriate solution for emulating what pam-auth-update can do (or maybe something else).
+  # Emulate in part what pam-auth-update does. There is a short coming right now, we
+  # cannot include other files from other slices that install into
+  #   /usr/share/pam-configs/*
+  # since this mutation script wont be able to access them as of writing.
   config:
+    essential:
+      - libpam-runtime_var
     contents:
       /etc/pam.conf:
+      /etc/pam.d/common-account: { text: "", mutable: true }
+      /etc/pam.d/common-auth: { text: "", mutable: true }
+      /etc/pam.d/common-password: { text: "", mutable: true }
+      /etc/pam.d/common-session: { text: "", mutable: true }
+      /etc/pam.d/common-session-noninteractive: { text: "", mutable: true }
       /etc/pam.d/other:
+      /usr/share/pam-configs/unix: { until: mutate }
+      /usr/share/pam/common-account: { until: mutate }
+      /usr/share/pam/common-auth: { until: mutate }
+      /usr/share/pam/common-password: { until: mutate }
+      /usr/share/pam/common-session: { until: mutate }
+      /usr/share/pam/common-session-noninteractive: { until: mutate }
+    mutate: |
+      def parse_type(t):
+        strippedValue = t[1].strip()
+        if t[0] == "Auth-Type":
+            return ["auth", strippedValue]
+        elif t[0] == "Account-Type":
+          return ["account", strippedValue]
+        elif t[0] == "Session-Type":
+          return ["session-noninteractive", strippedValue]
+        elif t[0] == "Password-Type":
+          return ["password", strippedValue]
+        return []
 
-  # folders expected by libpam to exist
-  var:
-    contents:
-      /var/lib/pam/:
+      confs_dir = "/usr/share/pam-configs/"
+      confs = content.list(confs_dir)
+      confdata = {}
+      for x in confs:
+        conf = content.read(confs_dir + x)
+        lines = conf.splitlines()
+
+        m = ""
+        p = ""
+        t = ""
+        for i in range(len(lines)):
+          vals = lines[i].split(":")
+          if vals[0] == "Priority":
+            p = vals[1].strip()
+          elif vals[0] in ["Auth-Type", "Account-Type", "Session-Type", "Password-Type"]:
+            m, t = parse_type(vals)
+          elif vals[0] == "Session-Interactive-Only":
+            if vals[1].strip() == "yes":
+              m = "session"
+          elif vals[0] in ["Auth", "Account", "Session", "Password"]:
```

Comment:
It seems that, in a clean container, we should use the values from the keys with `-Initial` suffices for these keys, e.g., `Auth-Initial`

Reply:
The remark seems reasonable to me. After inspecting existing usages of libpam (especially in containers like ubuntu/grafana), the -Initial configurations seem to be the default.